src/best_of/integrations/gitlab_integration.py

```diff
@@ -0,0 +1,212 @@
+import logging
+
+import requests
+from addict import Dict
+from dateutil.parser import parse
+
+from best_of import utils
+from best_of.default_config import MIN_PROJECT_DESC_LENGTH
+from best_of.integrations.base_integration import BaseIntegration
+
+log = logging.getLogger(__name__)
+
+query = """
+query($fullPath: ID!) {
+  project(fullPath: $fullPath) {
+    name
+    forksCount
+    starCount
+    issueStatusCounts {
+      all
+      closed
+      opened
+    }
+    description
+    createdAt
+    lastActivityAt
+    mergeRequests {
+      count
+    }
+    webUrl
+    httpUrlToRepo
+    statistics {
+      commitCount
+    }
+    releases(first: 100, sort: CREATED_DESC) {
+      edges {
+        node {
+          createdAt
+          name
+          releasedAt
+          tagName
+          tagPath
+          upcomingRelease
+        }
+      }
+    }
+  }
+}
+"""
+
+GITLAB_DEFAULT_API = "https://gitlab.com/api/graphql"
+
+
+class GitLabIntegration(BaseIntegration):
+    @property
+    def name(self) -> str:
+        return "gitlab"
+
+    def update_project_info(self, project_info: Dict) -> None:
+
+        # project_info:
+        # gitlab_url, homepage, name, gitlab_id, license, created_at
+        # updated_at, last_commit_pushed_at, fork_count, open_issue_count
+        # closed_issue_count, star_count, commit_count,
+        # latest_stable_release_published_at, latest_stable_release_number
+        # github_release_downloads, monthly_downloads, release_count
+        # description, dependent_project_count, contributor_count
+
+        api_url = project_info.api_url or GITLAB_DEFAULT_API
```

Comment:
Instead of adding a new `api_url` property, I would suggest to have the URL configured inside the `gitlab_id` property in case it is different from the default Gitlab API (similar how it is done with docker images). For example, if a different API endpoint that Gitlab default is needed: `<API_ENDPOINT>:org/repo`

Reply:
Good idea 👍